### DISCUSSION TITLE
Welcome to Discussions

### DISCUSSION THREAD

**Discussion by TorchDSP** [TorchDSP/torchsig#256] [Announcements]
To facilitate more open-source collaboration, we have created the Discussion space as well as a [Wiki](https://github.com/TorchDSP/torchsig/wiki).

**Comment by Maxwell20** [comment]
Was is the procedure to add a new data type to Wideband dataset? I would like to add CW spike and comb to the dataset and have the functions to do so. thanks!

**Reply by MattCarrickPL** [reply]
You need to implement a "class CWSpikeDataset" function in synthetic.py, refer to ConstellationDataset or FSKDataset for an example. https://github.com/TorchDSP/torchsig/blob/v0.6.0/torchsig/datasets/synthetic.py#L910

Add the class name into https://github.com/TorchDSP/torchsig/blob/v0.6.0/torchsig/datasets/signal_classes.py, in CLASS_FAMILY_DICT, 
SIGNALS_SHARED_LIST, FAMILY_SHARED_LIST, also suggested to fill out all of the appropriate logic as done for the other signal types in "class torchsig_signals"

The CWSpikeDataset class then needs to be integrated into ModulateNarrowbandDataset() here: https://github.com/TorchDSP/torchsig/blob/v0.6.0/torchsig/datasets/synthetic.py#L171, see also line 153. The signal can then be used in the narrowband dataset.

For wideband dataset support it needs to be added into wideband.py, for example here is how it is done for FSK: https://github.com/TorchDSP/torchsig/blob/v0.6.0/torchsig/datasets/wideband.py#L197

Please let me know if you have any more questions.

**Reply by MattCarrickPL** [reply]
If you'd like the code to be part of TorchSig, you're welcome to submit a pull request with fully or partially integrated code. If partially integrated, we can take a look to see what kind of time commitment it would be to integrate ourselves and may spend the time to do so.

**Comment by Maxwell20** [comment]
I would love to help extend torchsig to include some EMI types. Do I need
to be added as a developer on the prj?

**Reply by MattCarrickPL** [reply]
Create a fork, make the commits on the forked repo, and then send us a pull request

**Comment by Maxwell20** [comment]
sounds good

**Comment by Maxwell20** [comment]
also thank you for the detailed explanation

**Comment by rdaffey** [comment]
Are there plans to make TorchSig available on PyPI? I only ask as I'd like to use the tool at my organisation, but we are only permitted to download libraries from pypi.org.

**Reply by MattCarrickPL** [reply]
Interesting idea. I created an issue to track it. Discussing it internally. https://github.com/TorchDSP/torchsig/issues/341

**Reply by rdaffey** [reply]
Thank you!

**Reply by ereoh** [reply]
We are now live on PyPi. Check it our here: https://pypi.org/project/torchsig/

**Comment by randynovak** [comment]
It may be of interest to the community here to leverage Code2Flow (https://github.com/scottrogowski/code2flow). The call map of the TorchSig software is an interesting find. Requires a large plotter to print but the exported SVG file will do. If others have other offerings of interest, please share.

**Reply by ereoh** [reply]
Thanks for pointing it out, passing it to our developers.
